### ISSUE TITLE
X11 PRIME detection logic leaks memory according to Valgrind

### ISSUE THREAD

**Issue by qarmin** [godotengine/godot#29844]
**Godot version:**
3.2.dev.custom_build. 0d61fc2c0

**OS/device including version:**
Ubuntu 19.04, Gnome 3.32.1, GTX 970 Proprietary drivers 430.26

**Issue description:**
When I check Godot(now open only project manager, but same is with editor and game) with simple Valgrind 3.14 command then in log is shown a lot of leaked memory - [valgrindDefinite.txt](https://github.com/godotengine/godot/files/3296745/valgrindDefinite.txt)

```
valgrind --log-file="~/valgrind.txt" --leak-check=full --show-leak-kinds=definite godot
```
Also changing `definite` option to `all` increase number of possibly leaks - [valgrindALL.txt](https://github.com/godotengine/godot/files/3296733/valgrindALL.txt)

I'm not sure if that are Godot or hardware/Valgrind issue, so I tried to check Godot on my laptop with Intel HD 3000 and Mesa 19 but warnings are same as in issue #16214

**Comment by ibrahn** [comment]
The majority of the leaks logged are from a sub-process. (I think in this case, the one forked for PRIME gpu detection.)
This happens because we `exit` in the worker process and so the cleanup of allocations made before the fork doesn't happen on that side. Probably sensible, because it helps avoid  triggering a copy-on-write of pages that are otherwise shared with the parent process.
To ignore child processes, you can add `--child-silent-after-fork=yes` when running valgrind. Although this isn't much use if you're looking for a way of getting useful information on the child itself obviously.

**Comment by KoBeWi** [comment]
@qarmin Is this still valid in current master?

**Comment by qarmin** [comment]
For now no.
Godot 4.0 from which I know, doesn't have now PRIME detection, but adding that feature probably still will have leaks.

**Comment by KoBeWi** [comment]
> For now no.

So no point in keeping this open for now.

**Comment by akien-mga** [comment]
Well I'd keep this open for `3.2`, as it's valid there. And whatever fix is done for `3.2` should be applicable for `master` too, unless we drop `platform/linuxbsd/detect_prime_x11.cpp`.

**Comment by akien-mga** [comment]
Fixed by #56036.

**Comment by madmiraal** [comment]
This should be reopened, because #56036 was reverted.

**Comment by Rubonnek** [comment]
Fixed in #69400 with commit 5ada214c51e.
